src/main/java/xyz/nucleoid/extras/player_list/PlayerListHelper.java

```diff
@@ -1,15 +1,27 @@
 package xyz.nucleoid.extras.player_list;
 
+import net.minecraft.network.encryption.PlayerPublicKey;
 import net.minecraft.network.packet.s2c.play.PlayerListS2CPacket;
 import net.minecraft.server.network.ServerPlayerEntity;
 import net.minecraft.text.Text;
 import net.minecraft.util.Formatting;
 import net.minecraft.world.GameMode;
+import org.jetbrains.annotations.Nullable;
 import xyz.nucleoid.plasmid.game.manager.GameSpaceManager;
 
 public class PlayerListHelper {
     public static Text getDisplayName(ServerPlayerEntity player, boolean gray) {
-        return gray ? player.getName().shallowCopy().formatted(Formatting.DARK_GRAY, Formatting.ITALIC) : player.getDisplayName();
+        return gray ? player.getName().copy().formatted(Formatting.DARK_GRAY, Formatting.ITALIC) : player.getDisplayName();
+    }
+
+    @Nullable
+    private static PlayerPublicKey.PublicKeyData getPublicKey(ServerPlayerEntity player) {
+        var publicKey = player.getPublicKey();
+        if (publicKey == null) {
+            return null;
+        } else {
+            return publicKey.data();
+        }
```

Comment:
Some fun new Vanilla utils 😉 
```suggestion
        return Util.map(player.getPublicKey(), PlayerPublicKey::data);
```